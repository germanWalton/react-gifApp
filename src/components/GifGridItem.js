import React from 'react'

const GifGridItem = ({title,url}) => {


  return (
    <div className='card animate__animated animate__zoomIn'>
      <img className=""src={url} alt={title} />
      <p className="p-3">{title}</p>
    </div>
  )
}

export default GifGridItem