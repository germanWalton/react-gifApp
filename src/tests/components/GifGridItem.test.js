import React from 'react'
import { shallow } from 'enzyme'
import '@testing-library/jest-dom'
import GifGridItem from '../../components/GifGridItem'



describe('Pruebas en <GifGridItem />',()=> {

  const title = 'Goku';
  const url = 'http://www.google.com';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);


 test('debe mostrar el componente correctamente', () => {

   expect(wrapper).toMatchSnapshot();
 })
  
 test('debe mostrar un parrafo con el title', () => {

   const p = wrapper.find('p').text().trim();
   expect(p).toBe(title);
})

test('debe tener la imagen con la url y el alt de los props', () => {

  const img = wrapper.find('img');
  expect(img.prop('src')).toBe(url);
  expect(img.prop('alt')).toBe(title);
})
  
  
test('debe tener la clase animate__zoomIn', () => {

  const div = wrapper.find('div');
  const className = div.prop('className')
  expect(className.includes('animate__zoomIn')).toBe(true)
})

})