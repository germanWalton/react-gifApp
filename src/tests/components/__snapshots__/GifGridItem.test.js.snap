// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__zoomIn"
>
  <img
    alt="Goku"
    src="http://www.google.com"
  />
  <p
    className="p-3"
  >
    Goku
  </p>
</div>
`;
